packages/admin/admin/src/FinalForm.tsx

```diff
@@ -173,7 +173,7 @@ export function FinalForm<FormValues = AnyObject>(props: IProps<FormValues>) {
                         if (formRenderProps.hasValidationErrors) {
                             return false;
                         }
-                        const submissionErrors = await formRenderProps.form.submit();
+                        const submissionErrors = await handleSubmit(formRenderProps.values, formRenderProps.form);
```

Comment:
I can't believe that this doesn't break any existing form. I'd prefer changing the  `saveAction` to await validation, either by polling the form state as you do in waitForFormValidationToBeFinished or use a form subscriber (even cleaner IMO), and then call the normal `form.submit()` function (which only submits the form if it has no validation errors)

Reply:
The problem that still remains is that 

> `formRenderProps.form.submit();` doesn't actually return the submissionErrors and it doesn't wait for the submission to be finished before returning 
> (as I stated here https://github.com/vivid-planet/comet/pull/1353#discussion_r1376389244).

Meaning if the save button is clicked before validation is finished, the validation is awaited but no error is returned. So the EditDialog is still closed and a dirty handler dialog is shown.

Maybe we can avoid this for validation errors using a form subscriber. But if there is an actual submission error (in the API), we still get the dirty handler dialog.

Reply:
@johnnyomair I was wrong. Now I got it to work for submission and validations errors with `form.submit()`